backend/src/common/services/token/token.service.ts

```diff
@@ -0,0 +1,32 @@
+import {
+    type JWTPayload,
+    jwtVerify,
+    type JWTVerifyResult,
+    SignJWT,
+} from 'jose';
+
+import { config } from '~/common/config/config.js';
+
+import { type TokenService } from './interfaces/interface.js';
+
+class TokenServiceBase implements TokenService {
+    private secret: Uint8Array;
+
+    public constructor() {
+        this.secret = new TextEncoder().encode(config.ENV.JWT.SECRET);
+    }
+
+    public async create({ id }: JWTPayload): Promise<string> {
+        return await new SignJWT({ id })
+            .setProtectedHeader({ alg: config.ENV.JWT.ALG })
+            .setIssuedAt()
+            .setExpirationTime(config.ENV.JWT.EXPIRES_IN)
+            .sign(this.secret);
+    }
+
+    public async decode(token: string): Promise<JWTVerifyResult> {
+        return await jwtVerify(token, this.secret);
```

Comment:
I guess you need to remove await, because it's already return Promise
( https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/88#discussion_r1304760618 )
```suggestion
        return jwtVerify(token, this.secret);
```

Reply:
Done